include/sys/util.h

```diff
@@ -178,6 +178,16 @@ extern "C" {
 #define MIN(a, b) (((a) < (b)) ? (a) : (b))
 #endif
 
+/**
+ * @def CLAMP
+ * @brief Clamp a value to a given range.
+ * @note Arguments are evaluated multiple times.
+ */
+#ifndef CLAMP
+/* Use Z_CLAMP for a GCC-only, single evaluation version */
+#define CLAMP(val, low, high) MIN(MAX(val, low), high)
```

Comment:
Could we we use Z_CLAMP automatically if GCC is used?
```suggestion
#ifdef __GNUC__
/* Use Z_CLAMP for a GCC-only, single evaluation version */
#define CLAMP Z_CLAMP
#else
#define CLAMP(val, low, high) MIN(MAX(val, low), high)
#endif
```

Reply:
I don't think so, as it has limitations for compile time evaluation, as documented in `Z_MAX`. If we do this change for clamp, we should do it for min and max as well. I think that should be a separate PR in that case.

Reply:
Right, I see. Makes sense to keep it separate then.